hasher-matcher-actioner/hmalib/common/configs/tx_collab_config.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+DynamoDB backed CollaborationConfig objects.
+
+This may become a CollaborationConfigStoreBase, but since none of the pytx
+interfaces require it, keeping it as just another class for now.
+"""
+
+import typing as t
+import itertools
+
+from threatexchange.exchanges.collab_config import CollaborationConfigBase
+from threatexchange.exchanges.signal_exchange_api import SignalExchangeAPI
+
+from hmalib.common.config import HMAConfig, create_config
+
+
+def make_collab_config_subclass(
+    cls: t.Type[CollaborationConfigBase],
+) -> t.Type[HMAConfig]:
+    """
+    Generate an HMAConfig subclass on-the-fly.
+
+    Since supported APIs are dynamic, we need to allow any
+    CollaborationConfigBase to be stored as an HMAConfig object.
+    """
+    return type(f"_Dynamic{cls.__name__}_HMAConfig", (cls, HMAConfig), {})
```

Comment:
Oof, defintely not looking forward to debugging this in the future.

Why not 

```
  class _DynamicWhatever(cls, HMAConfig):
     pass
  _DynamicWhatever.__name__ = "name"

  return _DynamicWhatever
```

Which is slightly longer, but easier to spot what's happening

Reply:
I guess my Java instincts kicked in. I did not know you could use the class keyword to declare classes at runtime! TIL!